lib/executor/common_test.go

```diff
@@ -16,7 +16,7 @@ import (
 func setupExecutor(
 	t *testing.T, config lib.ExecutorConfig,
 	vuFn func(context.Context, chan<- stats.SampleContainer) error,
-) (context.Context, context.CancelFunc, lib.Executor, *testutils.SimpleLogrusHook) {
```

Comment:
I would argue (without going any further) that the ExecutionState should be a parameter ...

Reply:
Hhmm yeah, I can see how that would be cleaner. Thanks, will change.

Reply:
It won't only be cleaner
, but in a lot of places (as was said in my tests PR) we would want to test with different execution segments ... and it will be easier if we have a way to set them from now instead of having to refactor it again in the future.

Reply:
Resolved in 7684b3c6.